### PR TITLE
kubeadm operator: add unit test tests - part1

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR adds the first set of unit test for the kubeadm operator

### CODE REVIEW

File: operator/controllers/runtimetask_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,820 @@
...
+	type args struct {
+		task *operatorv1.RuntimeTask
+	}
+	type want struct {
+		task *operatorv1.RuntimeTask
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
```
Comment (neolit123) [inline operator/controllers/runtimetask_controller_test.go]: as mentioned on the @SataQiu PR this should be, like so:
https://github.com/kubernetes/kubeadm/pull/1923/files#diff-1175c7d3cbb2ee3af06c7f1bbd008d60R34-R36

File: operator/controllers/runtimetask_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,820 @@
...
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &RuntimeTaskReconciler{}
+			r.reconcilePhase(tt.args.task)
+
+			if !reflect.DeepEqual(tt.args.task, tt.want.task) {
+				t.Errorf("reconcilePhase() = %v, want %v", tt.args.task, tt.want.task)
```
Comment (neolit123) [inline operator/controllers/runtimetask_controller_test.go]: we should follow this pattern:
https://github.com/kubernetes/kubeadm/pull/1923/files#diff-1175c7d3cbb2ee3af06c7f1bbd008d60R134

including the function name is not needed.

File: operator/controllers/runtimetask_reconcile_test.go [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+			},
+		},
+	}
+
+	got := reconcileTasks(nodes, tasks)
+
+	// All
+	want := []string{"node1-a", "node1-b", "node1-c", "node1-d", "node2", "node3", "node4"}
+	if got := taskNames(got.all); !reflect.DeepEqual(got, want) {
+		t.Errorf("newTaskGroupChildProxy().all = %v, want %v", got, want)
```
Comment (neolit123) [inline operator/controllers/runtimetask_reconcile_test.go]: this list of "all", "pending"...etc, might be simplified with reflection and a table, but will be more difficult to read.

File: operator/controllers/runtimetask_reconcile_test.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "k8s.io/kubeadm/operator/api/v1alpha1"
+)
+
+func Test_newTaskGroupChildProxy(t *testing.T) {
```
Comment (neolit123) [inline operator/controllers/runtimetask_reconcile_test.go]: i think we shouldn't use underscores in function names

File: operator/controllers/runtimetask_controller_test.go (hunk 1 above)
Comment (fabriziopandini) [reply operator/controllers/runtimetask_controller_test.go]: [1] I don't see any value added in renaming args in input and want in output, so I don't think this should block the PR

TBH I have to say that the PR from @SataQiu deviated from the example provided, without discussing the point with the author of the issue

File: operator/controllers/runtimetask_controller_test.go (hunk 2 above)
Comment (fabriziopandini) [reply operator/controllers/runtimetask_controller_test.go]: I don't see any problem in having the name of the function in the error message; actually I think that having the function name can be useful in detecting and fixing the problem

File: operator/controllers/runtimetask_reconcile_test.go (hunk 3 above)
Comment (fabriziopandini) [reply operator/controllers/runtimetask_reconcile_test.go]: "all", "pending" etc are all part of the same struct, and all of them are set by the same call. Having a table will make it really difficult to read

File: operator/controllers/runtimetask_reconcile_test.go (hunk 4 above)
Comment (fabriziopandini) [reply operator/controllers/runtimetask_reconcile_test.go]: Same as [1] I don't see any value added in renaming, so I don't think this should block the PR

**Comment by neolit123** [comment]
leaving review and LGTM to @SataQiu 
/cc @SataQiu

**Comment by fabriziopandini** [comment]
@SataQiu, kind ping

**Comment by SataQiu** [comment]
I‘d like to merge this PR first. From a functional point of view, this PR is fine.
If you don't mind, I will send another PR after this merged to make all the test code have the same style.

/lgtm

**Comment by SataQiu** [comment]
@fabriziopandini 
The CI failed! This is because the function name conflicts (It's my fault).
I have sent a PR to resolve this issue #1942

**Comment by fabriziopandini** [comment]
@SataQiu 

> If you don't mind, I will send another PR after this merged to make all the test code have the same style.

Feel free to send such PR. My point is only to not block at this stage of this effort and to iterate quickly. Everything that goes in this direction is more than welcome!

**Pull request merged**